pkg/store/kotsstore/downstream_store.go

```diff
@@ -423,8 +427,10 @@ func (s *KOTSStore) GetDownstreamVersions(appID string, clusterID string, downlo
 		if err := s.AddDownstreamVersionDetails(appID, clusterID, v, false); err != nil {
 			return nil, errors.Wrap(err, "failed to add details to latest downloaded version")
 		}
-		v.IsDeployable, v.NonDeployableCause = isAppVersionDeployable(v, result, license.Spec.IsSemverRequired)
-		break
```

Comment:
i cant understand the reason for this break here but it makes it so that the loop only iterates for the first non-pending_download version

Reply:
the answer is in the comment above:
> // checking if a version is deployable requires getting all versions again.
> // check if latest version is deployable separately to avoid cycle dependencies between functions.

Reply:
populating if a version is deployable or not only happens in the version history handler because it's an expensive process.